v3/src/models/app-state.ts

```diff
@@ -120,13 +123,20 @@ class AppState {
         if (metadata) {
           const metadataEntries = Object.entries(metadata)
           metadataEntries.forEach(([key, value]) => {
-            if (value != null) {
+            if (value == null) return
+
+            if (key === "filename") {
+              // We don't save the filename because it is redundant with the filename in the actual
+              // filesystem.
+              // However we need to the extension-less name for the window title
```

Comment:
```suggestion
              // However we need the extension-less name for the window title.
```
or something like that.